channel/src/main/java/com/synopsys/integration/alert/channel/email2/EmailChannelMessageSender.java

```diff
@@ -0,0 +1,155 @@
+                   .orElseThrow(() -> new AlertConfigurationException("ERROR: Missing global config."));
+    }
+
+    private void sendMessage(EmailMessagingService emailService, EmailAttachmentFormat attachmentFormat, EmailChannelMessageModel message, Set<String> emailAddresses) throws AlertException {
+        HashMap<String, Object> model = new HashMap<>();
+        model.put(EmailPropertyKeys.EMAIL_CONTENT.getPropertyKey(), message.getContent());
+        model.put(EmailPropertyKeys.EMAIL_CATEGORY.getPropertyKey(), message.getMessageFormat());
+        model.put(EmailPropertyKeys.TEMPLATE_KEY_SUBJECT_LINE.getPropertyKey(), message.getSubjectLine());
+        model.put(EmailPropertyKeys.TEMPLATE_KEY_PROVIDER_URL.getPropertyKey(), message.getProviderUrl());
+        model.put(EmailPropertyKeys.TEMPLATE_KEY_PROVIDER_NAME.getPropertyKey(), message.getProviderName());
+        model.put(EmailPropertyKeys.TEMPLATE_KEY_PROVIDER_PROJECT_NAME.getPropertyKey(), message.getProjectName().orElse("Global"));
+        model.put(EmailPropertyKeys.TEMPLATE_KEY_START_DATE.getPropertyKey(), String.valueOf(System.currentTimeMillis()));
+        model.put(EmailPropertyKeys.TEMPLATE_KEY_END_DATE.getPropertyKey(), String.valueOf(System.currentTimeMillis()));
+        model.put(FreemarkerTemplatingService.KEY_ALERT_SERVER_URL, alertProperties.getServerUrl().orElse("#"));
+
+        Map<String, String> contentIdsToFilePaths = new HashMap<>();
+        emailService.addTemplateImage(model, contentIdsToFilePaths, EmailPropertyKeys.EMAIL_LOGO_IMAGE.getPropertyKey(), getImagePath(FILE_NAME_SYNOPSYS_LOGO));
+
+        EmailTarget emailTarget = new EmailTarget(emailAddresses, FILE_NAME_MESSAGE_TEMPLATE, model, contentIdsToFilePaths);
+        Optional<File> optionalAttachment = message.getSource().flatMap(projectMessage -> addAttachment(emailTarget, attachmentFormat, projectMessage));
+
+        emailService.sendEmailMessage(emailTarget);
+        optionalAttachment.ifPresent(emailAttachmentFileCreator::cleanUpAttachmentFile);
+    }
+
+    private Optional<File> addAttachment(EmailTarget emailTarget, EmailAttachmentFormat attachmentFormat, ProjectMessage projectMessage) {
+        Optional<File> optionalAttachmentFile = emailAttachmentFileCreator.createAttachmentFile(attachmentFormat, projectMessage);
+        if (optionalAttachmentFile.isPresent()) {
+            File attachmentFile = optionalAttachmentFile.get();
+            // We trust that the file was created correctly, so the path should be correct.
+            emailTarget.setAttachmentFilePath(attachmentFile.getPath());
+        }
+        return optionalAttachmentFile;
+    }
+
+    private String getImagePath(String imageFileName) throws AlertException {
+        String imagesDirectory = alertProperties.getAlertImagesDir();
+        if (StringUtils.isNotBlank(imagesDirectory)) {
+            return imagesDirectory + "/" + imageFileName;
```

Comment:
Should we be using some sort of path builder to build these paths instead of strings? I think you can use Paths to merge these together.

Reply:
Let's add it as an improvement ticket internally. I'm trying to stay as true as possible (i.e. copy/paste) to the same functionality as before in this PR.